cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/SparkCqlEvaluator.java

```diff
@@ -760,6 +777,29 @@ protected void writeResults(SparkSession spark, StructType schema, JavaPairRDD<O
                 .save(outputURI);
     }
 
+    private Map<String, String> createDataTypeAliases(List<ContextDefinition> filteredContexts, CqlToElmTranslator translator) {
+        Set<String> dataTypes = filteredContexts.stream().map(ContextDefinition::getPrimaryDataType).collect(Collectors.toSet());
+        Collection<ModelInfo> modelInfos = translator.getRegisteredModelInfos().values();
+
+        Map<String, String> dataTypeAliases = new HashMap<>();
+        for (ModelInfo modelInfo : modelInfos) {
+            modelInfo.getTypeInfo().stream().map(ClassInfo.class::cast)
+                .filter(classInfo -> dataTypes.contains(classInfo.getName()))
+                .forEach(info -> {
+                    String dataType = info.getName();
+                    QName baseType = ModelUtils.getBaseTypeName(modelInfo, info);
+                    if (baseType != null) {
+                        dataTypeAliases.put(dataType, baseType.getLocalPart());
+                    }
+
+                    Collection<String> choiceTypes = ModelUtils.getChoiceTypeNames(info);
+                    choiceTypes.forEach(choiceType -> dataTypeAliases.put(dataType, choiceType));
```

Comment:
Pedantic: It might be worth adding a simple comment above the two "blocks" of code in here.
Something like `// for inheritance types` and `// for choice types`.
I just want the viewer to know that the two blocks are kinda unrelated and serve different use cases.
This is similar to how you commented in `SparkSchemaCreator`.

Reply:
addressed in 04ac3ec